packages/react/src/createReactQueryHooks.tsx

```diff
@@ -309,7 +309,18 @@ export function createReactQueryHooks<
       TError
     >,
   ): UseQueryResult<TData, TError> {
-    const { client, ssrState, queryClient, prefetchQuery } = useContext();
+    const { client, ssrState, queryClient, prefetchQuery, cancelQuery } =
+      useContext();
+
+    React.useEffect(() => {
+      const abortQuery = () => {
+        void cancelQuery([pathAndInput[0]]);
+      };
+      opts?.trpc?.signal?.addEventListener('abort', abortQuery);
+      return () => {
+        opts?.trpc?.signal?.removeEventListener('abort', abortQuery);
+      };
+    });
```

Comment:
React Query [passes an abort signal to each query](https://tanstack.com/query/v4/docs/guides/query-cancellation) so we should use that instead. We also shouldn't allow passing a signal in the options here since you should just be calling `utils.post.all.cancel()`